tiledb/ml/data_apis/tensorflow_sparse.py

```diff
@@ -0,0 +1,158 @@
+"""Functionality for loading data directly from TileDB arrays into the Tensorflow Data API."""
+import dataclasses
+
+import tiledb
+import tensorflow as tf
+import numpy as np
+from tensorflow.python.framework import constant_op
+from tensorflow.python.data.ops import dataset_ops
+
+
+class TensorflowTileDBSparseDataset(tf.data.Dataset):
+    """
+    Class that implements all functionality needed to load sparse data from TileDB directly to the
+    Tensorflow Data API, by employing generators.
+    """
+
+    def __new__(cls, x_array: tiledb.Array, y_array: tiledb.Array, batch_size: int):
+        """
+        Returns a Tensorflow Dataset object which loads data from TileDB arrays by employing a generator.
+        :param x_array: TileDB Sparse Array. Array that contains features.
+        :param y_array: TileDB Dense/Sparse Array. Array that contains labels.
+        :param batch_size: Integer. The size of the batch that the implemented _generator method will return.
+        For optimal reads from a TileDB array, it is recommended to set the batch size equal to the tile extent of the
+        dimension we query (here, we always query the first dimension of a TileDB array) in order to get a slice (batch)
+        of the data. For example, in case the tile extent of the first dimension of a TileDB array (x or y) is equal to
+        32, it's recommended to set batch_size=32. Any batch size will work, but in case it's not equal the tile extent
+        of the first dimension of the TileDB array, you won't achieve highest read speed. For more details on tiles,
+        tile extent and indices in TileDB, please check here:
+        https://docs.tiledb.com/main/solutions/tiledb-embedded/performance-tips/choosing-tiling-and-cell-layout#dense-arrays
+        """
+        cls.x = x_array
+        cls.y = y_array
+
+        # Check that x and y have the same number of rows
+        if x_array.schema.domain.shape[0] != y_array.schema.domain.shape[0]:
+            raise Exception(
+                "X and Y should have the same number of rows, i.e., the 1st dimension "
+                "of TileDB arrays X, Y should be of equal domain extent."
+            )
+
+        # Get number of observations
+        rows = x_array.schema.domain.shape[0]
+
+        # Get x and y shapes
+        cls.x_shape = (None,) + x_array.schema.domain.shape[1:]
```

Comment:
This won't be the true shape in most cases for sparse. Does this matter?

Reply:
@Shelnutt2 Can you clarify true shape? Do you mean in case the sparse domains are of `float` or `string` dtype?